src/lib/services/nft/index.ts

```diff
@@ -299,16 +301,44 @@ export const useNftMintInfo = (nftAddress: HexAddr32) => {
   );
 };
 
-export const useMetadata = (uri: string) =>
-  useQuery<Metadata>(
-    [CELATONE_QUERY_KEYS.NFT_METADATA, uri],
-    () => getMetadata(uri),
+export const useMetadata = (
+  nft: Option<Partial<Nft>>,
+  width = "",
+  height = ""
+) => {
+  const { currentChainId } = useCelatoneApp();
+
+  return useQuery<Metadata>(
+    [CELATONE_QUERY_KEYS.NFT_METADATA, nft],
+    async () => {
+      if (!nft) throw new Error("NFT is required (useMetadata)");
+      const baseUri = await getMetadata(nft.uri ?? "");
+
+      if (
+        baseUri.image.startsWith("ipfs://") &&
+        nft.collectionAddress &&
+        (nft.nftAddress || nft.tokenId)
+      ) {
+        const image: Blob = await getGlyphImage(
+          currentChainId,
+          nft.collectionAddress,
+          nft.nftAddress ?? nft.tokenId ?? "",
+          width,
+          height
+        );
+
+        baseUri.image = URL.createObjectURL(image);
```

Comment:
I think we should do a try catch here. If glyph doesn't work, we fallback to `getIpfsUrl` here

Then we don't have to check ipfs url when using it anymore